pkg/integrations/v2/app_agent_receiver/handler.go

```diff
@@ -0,0 +1,124 @@
+package app_agent_receiver
+
+import (
+	"context"
+	"sync"
+
+	"crypto/subtle"
+	"encoding/json"
+	"net/http"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/rs/cors"
+	"golang.org/x/time/rate"
+)
+
+type appAgentReceiverExporter interface {
+	Name() string
+	Export(ctx context.Context, payload Payload) error
+}
+
+// AppAgentReceiverHandler struct controls the data ingestion http handler of the receiver
+type AppAgentReceiverHandler struct {
+	exporters               []appAgentReceiverExporter
+	config                  *Config
+	rateLimiter             *rate.Limiter
+	exporterErrorsCollector *prometheus.CounterVec
+}
+
+// NewAppAgentReceiverHandler creates a new AppReceiver instance based on the given configuration
+func NewAppAgentReceiverHandler(conf *Config, exporters []appAgentReceiverExporter, reg *prometheus.Registry) AppAgentReceiverHandler {
```

Comment:
nit: pass around `prometheus.Registrar` instead of `*prometheus.Registry` to allow for different implementations (e.g., prometheus.WrapRegistrarWith, etc)

Reply:
- [ ] use`prometheus.Registrar`